src/DIRAC/Resources/Computing/AREXComputingElement.py

```diff
@@ -0,0 +1,668 @@
+        self.pilotProxy = ""
+        self.queue = ""
+        self.ceHost = self.ceName
+
+        result = returnValueOrRaise(getCESiteMapping(self.ceHost))
+        self.site = result[self.ceHost]
+        self.grid = self.site.split(".")[0]
+
+        # For the REST interface
+        if self.initialised:  # a bunch of the local variables are set up here.
+            self.log.info("REST interface successfully configured", "for %s" % self.ceHost)
+        else:
+            self.log.info("REST interface not yet properly configured", "for %s" % self.ceHost)
+            self.log.info("Try again in next iteration", "Bye Bye")
+            return S_ERROR("Try try try initialisation again")
+
+        # Do these after all other initialisations, in case something barks
+        self.xrslExtraString = self.__getXRSLExtraString()
+        self.xrslMPExtraString = self.__getXRSLExtraString(multiprocessor=True)
+
+        # Get from the CS the time left (should be in seconds) for proxy before renewal
+        self.proxyTimeLeftBeforeRenewal = 10000  # 2 hours, 46 minutes and 40 seconds
+        variable = "proxyTimeLeftBeforeRenewal"
+        firstOption = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, variable)
+        secondOption = cfgPath("Resources", "Sites", self.grid, self.site, variable)
+        defaultOption = cfgPath("Resources", "Computing" "CEDefaults", variable)
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
+        if result["OK"]:
+            self.proxyTimeLeftBeforeRenewal = result["Value"]
+
+        self.arcRESTTimeout = 1.0
+        variable = "ARCRESTTimeout"
+        firstOption = cfgPath("Resources", "Sites", self.grid, self.site, "CEs", self.ceHost, variable)
+        secondOption = cfgPath("Resources", "Sites", self.grid, self.site, variable)
+        defaultOption = cfgPath("Resources", "Computing" "CEDefaults", variable)
+        result = self._getFromCfgSvc(firstOption, secondOption, defaultOption)
+        if result["OK"]:
+            self.arcRESTTimeout = result["Value"]
+
+    #############################################################################
+    @property
+    def initialised(self):
+        """Check if the REST interface is available and ready, configuring it if required.
+        Hopefully this is executed just once per CE, though it should not harm to run it more often.
+        If all is well the following variables are set, and a few others on which they (especially self.session)
+        depend.
+        self.initialised : Static variable to say if the REST interface is available and ready
+        self.base_url : The basic URL for the REST interface
+        self.session  : The python "requests" module session
+        self.headers  : The format of information we prefer : json of course (except during delegations)
+        Specification : https://www.nordugrid.org/arc/arc6/tech/rest/rest.html
+        """
+        if hasattr(self, "s"):
```

Comment:
I guess this should be:
```suggestion
        if hasattr(self, "session"):
```

Reply:
Thanks!